src/cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -122,6 +125,22 @@ static void set_program_name_in_venv(void) {
 extern "C" COCOTB_EXPORT void _embed_init_python(void) {
     assert(!gtstate);  // this function should not be called twice
 
+    const char* log_level = getenv("COCOTB_LOG_LEVEL");
+    if (log_level){
+        static const std::map<std::string, int> logStrToLevel = {
+            {"CRITICAL", GPICritical},
+            {"ERROR", GPIError},
+            {"WARNING", GPIWarning},
+            {"INFO", GPIInfo},
+            {"DEBUG", GPIDebug},
+            {"TRACE", GPITrace}
+        };
+        auto it = logStrToLevel.find(log_level);
+        if (it != logStrToLevel.end()) {
+            py_gpi_logger_set_level(it->second);
+        }
```

Comment:
Maybe add an else here with a message stating that the log level wasn't valid.

Reply:
It has a decent message in log.py: 
```  File "/usr/scratch/anolte/local/cocos/cocotb/src/cocotb/log.py", line 104, in default_config
    raise ValueError(
ValueError: Invalid log level 'INF' passed through the COCOTB_LOG_LEVEL environment variable. Valid log levels: CRITICAL, ERROR, WARNING, INFO, DEBUG, TRACE
```
Not sure if we want to duplicate this